### PR TITLE
rgw: add librgw acl interface

### PR DESCRIPTION
**Pull request by platoxie**
Signed-off-by: Yiming Xie <plato@umcloud.com>

### CODE REVIEW

**Comment by platoxie** [comment]
@yehudasa please review it when you are free. Thank you.

File: src/rgw/rgw_file.cc [hunk 1]
```diff
@@ -377,7 +377,7 @@ namespace rgw {
   } /* RGWLibFS::unlink */
 
   int RGWLibFS::rename(RGWFileHandle* src_fh, RGWFileHandle* dst_fh,
-		       const char *_src_name, const char *_dst_name)
+		       const char *_src_name, const char *_dst_name, const char* acl="")
```
Comment (ZVampirEM77) [inline src/rgw/rgw_file.cc]: Can we define the default value of parameter `acl` in function `rename` declaration in rgw_file.h? It should not be here.

File: src/rgw/rgw_file.cc [hunk 2]
```diff
@@ -496,7 +496,7 @@ namespace rgw {
   } /* RGWLibFS::rename */
 
   MkObjResult RGWLibFS::mkdir(RGWFileHandle* parent, const char *name,
-			      struct stat *st, uint32_t mask, uint32_t flags)
+			      struct stat *st, uint32_t mask, uint32_t flags, const char* acl="")
```
Comment (ZVampirEM77) [inline src/rgw/rgw_file.cc]: same

**Comment by mattbenjamin** [comment]
@platoxie broadly, what is the application that will be consuming this?  I guess not NFS.  I guess it makes sense to expose acls, but what about policy &c?

I'm not sure that I like the new external api method names;  still thinking about it

For sure, though, you need to increment the api minor version in include/rados/rgw_file.h whenever it changes.

**Comment by platoxie** [comment]
@mattbenjamin Thanks for your suggest！
we use ceph as the storage backend for streaming media. To support push & pull stream, we need to have some control over the buckets and object acl via librgw.
1.what do you mean about "policy &c?"  
2. `increment the api minor version in include/rados/rgw_file.h whenever it changes` you mean to change LIBRGW_FILE_VER_MINOR from 1 to 2 ?

**Comment by mattbenjamin** [comment]
@platoxie re LIBRGW_FILE_VER, I think this would be a minor version change, yes;

**Comment by mattbenjamin** [comment]
@platoxie re the new api methods:  rather than adding "with_acl" variations of several methods, given that we are revving the API version, I think we an just add the ACL argument, which may be NULL, which in your implementation will work as expected;  My comment about policy was, our API doesn't provide a way to set AWS bucket policy, either--for that, I'm inclined to say we do not need an 'atomic' interface to set on creation, but we could expose methods to set policy explicitly?

**Comment by mattbenjamin** [comment]
@adamemerson could you review and comment?

**Comment by mattbenjamin** [comment]
@platoxie are you still interested in pursuing this?

**Pull request closed without merging**